pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -308,17 +308,10 @@ def _prepare_outputs(
                 opt_outputs = [opt_outputs]
 
             for batch_outputs in opt_outputs:
-                processed_tbptt_outputs = []
-
-                if isinstance(batch_outputs, ClosureResult):
+                if isinstance(batch_outputs, OutputResult):
                     batch_outputs = [batch_outputs]
 
-                for tbptt_output in batch_outputs:
-                    out = {}
-                    if tbptt_output.loss is not None:
-                        out["loss"] = tbptt_output.loss
-                    out.update(tbptt_output.extra)
-                    processed_tbptt_outputs.append(out)
+                processed_tbptt_outputs = [tbptt_output.extra for tbptt_output in batch_outputs]
```

Comment:
Broken, needs to be fixed.

We need different logic depending on the result class. Either that or a common interface to return what's processed.

Reply:
the training batch loop owns the optimizer loop and the manual loop. therefore it can be the training_batch_Loop's responsibility to retrieve the output and standardize it into a format that the higher-up loop expects. what do you think?

Reply:
I think we should split `_prepare_outputs` into pieces across loops. Right now the information of the subloops leaks into the epoch loop as this function tries to process everything at once.

If you have ideas around it, I'm all ears 😄